### PR TITLE
New compat v19 php8.2

### PR DESCRIPTION
**Pull request by atm-jonathan**

### CODE REVIEW

File: backport/v19/core/class/commonhookactions.class.php [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+/**
+ *	\file       htdocs/core/class/commonhookactions.class.php
+ *	\ingroup    core
+ *	\brief      File of parent class of all other hook actions classes
+ */
+
+
+if (file_exists(DOL_DOCUMENT_ROOT . '/htdocs/core/class/commonhookactions.class.php')){
+
+	require_once DOL_DOCUMENT_ROOT . '/htdocs/core/class/commonhookactions.class.php';
```
Comment (atm-quentin) [inline backport/v19/core/class/commonhookactions.class.php:29]:
```suggestion
if (file_exists(DOL_DOCUMENT_ROOT . '/core/class/commonhookactions.class.php')){

	require_once DOL_DOCUMENT_ROOT . '/core/class/commonhookactions.class.php';
```

File: class/actions_free2product.class.php [hunk 2]
```diff
@@ -72,7 +71,7 @@ function addMoreActionsButtons($parameters, &$object, &$action, $hookmanager){
 			
 			global $addButtonToConvertAll;
 			
-			$base_new_ref = !empty($conf->global->FREE2PRODUCT_BASE_NEWREF) ? $conf->global->FREE2PRODUCT_BASE_NEWREF : 'FREELINE-';
+			$base_new_ref = getDolGlobalString('FREE2PRODUCT_BASE_NEWREF') ?? 'FREELINE-';
```
Comment (atm-quentin) [inline class/actions_free2product.class.php:74]:
```suggestion
			$base_new_ref = getDolGlobalString('FREE2PRODUCT_BASE_NEWREF, 'FREELINE-');
```

File: class/actions_free2product.class.php [hunk 3]
```diff
@@ -137,7 +136,7 @@ function formObjectOptions($parameters, &$object, &$action, $hookmanager)
 		  	
 			global $langs,$conf;
 			$langs->load('free2product@free2product');
-			$base_new_ref = !empty($conf->global->FREE2PRODUCT_BASE_NEWREF) ? $conf->global->FREE2PRODUCT_BASE_NEWREF : 'FREELINE-';
+			$base_new_ref = getDolGlobalString('FREE2PRODUCT_BASE_NEWREF') ?? 'FREELINE-';
```
Comment (atm-quentin) [inline class/actions_free2product.class.php:139]:
```suggestion
			$base_new_ref = getDolGlobalString('FREE2PRODUCT_BASE_NEWREF', 'FREELINE-');
```

File: script/interface.php [hunk 4]
```diff
@@ -17,9 +17,9 @@
 			$product_type = GETPOST('product_type');
 			$tva = GETPOST('tva');
 			
-			$base_new_ref = !empty($conf->global->FREE2PRODUCT_BASE_NEWREF) ? $conf->global->FREE2PRODUCT_BASE_NEWREF : 'FREELINE-';
+			$base_new_ref = getDolGlobalString('FREE2PRODUCT_BASE_NEWREF') ?? 'FREELINE-';
```
Comment (atm-quentin) [inline script/interface.php:20]:
```suggestion
			$base_new_ref = getDolGlobalString('FREE2PRODUCT_BASE_NEWREF', 'FREELINE-');
```

File: class/actions_free2product.class.php [hunk 5]
```diff
@@ -174,11 +173,7 @@ function convertAllFreeLine() {
 					
 					function redirect()
 					{
-						document.location.href="<?php
-						if(DOL_VERSION>=6 && $object->element == 'propal') echo dol_buildpath('/comm/propal/card.php?id='.$object->id,1);
-						else if($object->element == 'propal') echo dol_buildpath('/comm/propal.php?id='.$object->id,1);
-						else if($object->element == 'commande') echo dol_buildpath('/commande/card.php?id='.$object->id,1);
-						?>";
+						document.location.href='';
```
Comment (atm-john) [inline class/actions_free2product.class.php:176]: What ?

File: class/actions_free2product.class.php [hunk 6]
```diff
@@ -22,11 +22,11 @@
  * \brief   This file is an example hook overload class file
  *          Put some comments here
  */
-
+require_once __DIR__ . '/../backport/v19/core/class/commonhookactions.class.php';
 /**
  * Class ActionsFree2Product
  */
-class ActionsFree2Product
+class ActionsFree2Product extends \fraisdeport\RetroCompatCommonHookActions
```
Comment (atm-quentin) [inline class/actions_free2product.class.php:29]: fraisdeport

**Pull request merged**